cmd/otelcol/config/collector/config.d.linux/receivers/envoy.discovery.yaml

```diff
@@ -0,0 +1,39 @@
+#####################################################################################
+# This file is generated by the Splunk Distribution of the OpenTelemetry Collector. #
+#                                                                                   #
+# It reflects the default configuration bundled in the Collector executable for use #
+# in discovery mode (--discovery) and is provided for reference or customization.   #
+# Please note that any changes made to this file will need to be reconciled during  #
+# upgrades of the Collector.                                                        #
+#####################################################################################
+# prometheus:
+#   enabled: true
+#   rule:
+#     docker_observer: type == "container" and any([name, image, command], {# matches "(?i)envoy"}) and not (command matches "splunk.discovery")
+#     host_observer: type == "hostport" and command matches "(?i)envoy" and not (command matches "splunk.discovery")
+#     k8s_observer: type == "port" and pod.name matches "(?i)envoy"
```

Comment:
Is the k8s rule intentionally matching just pod name instead of the broader rule used by docker_observer? Istio sidecars for e.g. have pod name like `istio-proxy`, the image name might have envoy in it. But maybe we should only match standalone envoy to startwith

Reply:
Right, you had a note about that. I think we can update the rule to add istio-proxy but then I really need a test showing this works for an istio mesh.

Reply:
I am tempted to make a separate integration test with a kind cluster, we deploy everything and we make sure it works. I might get into that next and tweak the rule based on that.

Reply:
I think we should merge this one as-is. And have a different item to officially support istio. It is not just the pod name; [default prom port used by istio](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging) is also not same as the one here. Don't know if it is supported in deiscover mode, but it would be better  getting the port/path from the prometheus.io annotations for istio (like we have in our helm chart).